lib/ui/screens/contact_list_screen.dart

```diff
@@ -105,30 +118,34 @@ class _ContactListScreenState extends State<ContactListScreen> {
                   onPressed: _gotoAddNewContact)
             ],
           ),
-          body: Container(
-              padding: EdgeInsets.all(20),
-              child: ValueListenableBuilder(
+          body: Column(
+              children: [
+                SearchBox(onChanged: _onChanged),
+                ValueListenableBuilder(
                   valueListenable: ContactsRepository.getBox().listenable(),
                   builder: (context, Box contactsBox, _) {
-                    if (ContactsRepository.getBox().isNotEmpty)
-                      return ListView.builder(
-                          itemCount: contactsBox.length,
-                          itemBuilder: (context, index) {
-                            final Contact ? contact = ContactsRepository.getContactInfo(index);
-                            if (contact != null){
-                              String contactName = contact.firstName + " " +
-                                  contact.lastName;
-                              String phone = contact.phoneNumber;
-                              return Card(
-                                child: contactTile(contactName, phone, index),
-                              );
+                    if (filteredContactList.isNotEmpty)
```

Comment:
Better if create ValueNotifier<List<Contact>> _filteredContactListNotifier and use ValueListenableBuilder here to rebuild the list.
setState() function will cause whole widget will be rebuild, that's not necessary and losing performance.